app/src/colorpalettewidget.cpp

```diff
@@ -71,20 +72,28 @@ void ColorPaletteWidget::initUI()
                              "QPushButton:pressed { border: 1px solid #ADADAD; border-radius: 2px; background-color: #D5D5D5; }"
                              "QPushButton:checked { border: 1px solid #ADADAD; border-radius: 2px; background-color: #D5D5D5; }";
 
+
+    // color number need to be set to add new color to palette
+    // in cases where the last color was saved
+    // otherwise first color will be black.
+    editor()->color()->setColorNumber(0);
+    editor()->color()->setColor(editor()->color()->frontColor());
+
     ui->addColorButton->setStyleSheet(buttonStylesheet);
     ui->removeColorButton->setStyleSheet(buttonStylesheet);
     ui->colorDialogButton->setStyleSheet(buttonStylesheet);
 
     palettePreferences();
 
-    connect(ui->colorListWidget, &QListWidget::currentItemChanged, this, &ColorPaletteWidget::colorListCurrentItemChanged);
     connect(ui->colorListWidget, &QListWidget::itemClicked, this, &ColorPaletteWidget::clickColorListItem);
     connect(ui->colorListWidget, &QListWidget::itemDoubleClicked, this, &ColorPaletteWidget::changeColourName);
     connect(ui->colorListWidget, &QListWidget::currentTextChanged, this, &ColorPaletteWidget::onActiveColorNameChange);
 
     connect(ui->addColorButton, &QPushButton::clicked, this, &ColorPaletteWidget::clickAddColorButton);
     connect(ui->colorDialogButton, &QPushButton::clicked, this, &ColorPaletteWidget::clickColorDialogButton);
     connect(ui->removeColorButton, &QPushButton::clicked, this, &ColorPaletteWidget::clickRemoveColorButton);
+    connect(ui->colorListWidget, &QListWidget::itemSelectionChanged, this, &ColorPaletteWidget::selectedItems);
```

Comment:
Is this connect() necessary? in `selectedItems()` it seems doing nothing.

Reply:
Ah that's a remnant from earlier testing, will remove